kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/schemaextractor/ExtractSchemaRule.kt

```diff
@@ -319,7 +322,50 @@ class SchemaExtractor(
     }
   }
 
-  private fun extractVerb(verb: KtNamedFunction): Verb {
+  private data class ExportAnnotation(
+    val visibility: Visibility?,
+    val ingress: Ingress?,
+    val method: Method?,
+    val path: String?
+  )
+
+  private fun extractExportAnnotation(verb: KtNamedFunction, bindingContext: BindingContext): ExportAnnotation {
```

Comment:
nit- i think this can be accomplished without any member data and so it can be a static function declared in the `companion object` below, like the following:
```
private fun KtAnnotationEntry.toExportAnnotation(): ExportAnnotation {
      var visibility: Visibility? = null
      var ingress: Ingress? = null
      var method: Method? = null
      var path: String? = null
      exportAnnotation?.valueArguments?.forEach { arg ->
        when {
          arg.getArgumentName()?.asName?.asString() == "visibility" || arg.getArgumentExpression()?.getType(bindingContext)?.fqNameOrNull()?.asString() == Visibility::class.qualifiedName -> {
            visibility = kotlin.runCatching { Visibility.valueOf(arg.getArgumentExpression()?.text?.substringAfterLast('.') ?: "") }.getOrNull()
          }
          arg.getArgumentName()?.asName?.asString() == "ingress" || arg.getArgumentExpression()?.getType(bindingContext)?.fqNameOrNull()?.asString() == Ingress::class.qualifiedName -> {
            ingress = kotlin.runCatching { Ingress.valueOf(arg.getArgumentExpression()?.text?.substringAfterLast('.') ?: "") }.getOrNull()
          }
          arg.getArgumentName()?.asName?.asString() == "method" || arg.getArgumentExpression()?.getType(bindingContext)?.fqNameOrNull()?.asString() == Method::class.qualifiedName -> {
            method = kotlin.runCatching { Method.valueOf(arg.getArgumentExpression()?.text?.substringAfterLast('.') ?: "") }.getOrNull()
          }
          arg.getArgumentName()?.asName?.asString() == "path" || arg.getArgumentExpression()?.getType(bindingContext)?.fqNameOrNull()?.asString() == String::class.qualifiedName -> {
            path = arg.getArgumentExpression()?.text?.trim('"')
          }
        }
      }
      return ExportAnnotation(visibility, ingress, method, path)
    }
```    
then to use it, you'd do `annotationEntry.ToExportAnnotation()`